### PR TITLE
Add message to submitted page

### PR DESCRIPTION
**Pull request by DavidBiddle**
### What problem does this pull request solve?

Trello card: https://trello.com/c/ZRNnzzX2/1132-update-to-final-your-form-has-been-submitted-page-content-to-inform-user-to-expect-an-email

Conditionally adds text saying 'We have sent you a confirmation email.' to the confirmation page if the user has opted to receive an email confirmation. This is implemented using a new custom flash message, `email_sent`, which is passed to the view via the redirect from the `submit_answers` action.

Also includes two small fixes:
- A small update to the email confirmation field's hint text content to reflect the latest designs.
- A small change to improve how we check the session to see if the form has been submitted, and improve consistency when retrieving form answers (hat-tip to @aliuk2012). This fixes an issue which caused some confusing test failures with this PR.

#### When user has opted into the email
A large green panel component with the text 'Your form has been submitted', followed by a short paragraph saying'We have sent you a confirmation email.', followed by a heading 'What happens next', followed by a paragraph that reads 'We'll send you an email to let you know the outcome. You'll usually get a response within 10 working days.'

#### When user has not opted into the email
A large green panel component with the text 'Your form has been submitted', followed by a heading 'What happens next', followed by a paragraph that reads 'We'll send you an email to let you know the outcome. You'll usually get a response within 10 working days.'

### Things to consider when reviewing

- Ensure that you consider the wider context.
- Does it work when run on your machine?
- Is it clear what the code is doing?
- Do the commit messages explain why the changes were made?
- Are there all the unit tests needed?
- Has all relevant documentation been updated?

### CODE REVIEW

File: app/views/forms/submitted/submitted.html.erb [hunk 1]
```diff
@@ -4,8 +4,12 @@
   <div class="govuk-grid-column-two-thirds">
     <%= govuk_panel(title_text: t('form.submitted.title')) %>
 
+    <% if show_email_message %>
+      <p>We have sent you a confirmation email.</p>
+    <% end %>
+
     <%if @current_context.form.what_happens_next_text.present? %>
-      <h2 class="govuk-heading-m govuk-!-margin-top-7">What happens next</h2>
```
Comment (aliuk2012) [inline app/views/forms/submitted/submitted.html.erb:8]: why are we removing the margin?

Comment (DavidBiddle) [reply app/views/forms/submitted/submitted.html.erb:8]: It made the spacing weirdly large when the 'email sent' message is present. I've now moved the spacing from the top of the h2 to the bottom of the panel component, so the spacing should look good in both variations now.

File: app/controllers/forms/check_your_answers_controller.rb [hunk 2]
```diff
@@ -22,7 +22,13 @@ def submit_answers
           FormSubmissionService.call(current_context:,
                                      reference: email_confirmation_form.notify_reference,
                                      preview_mode: mode.preview?).submit
-          redirect_to :form_submitted
+
+          if email_confirmation_form.send_confirmation == "send_email"
```
[the remark this answers is not in the document]
Comment (DavidBiddle) [reply app/controllers/forms/check_your_answers_controller.rb]: Is that something the service currently returns, or something we'd like to add in a future PR?

[the remark this answers is not in the document]
Comment (aliuk2012) [reply app/controllers/forms/check_your_answers_controller.rb]: I think a future PR

File: spec/features/email_confirmation_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    expect(page.find("h1")).to have_text I18n.t("form.submitted.title")
+    expect(page).to have_text I18n.t("form.submitted.email_sent")
+  end
+
+  def fill_in_form
+    visit form_path(mode: "form", form_id: 1, form_slug: "apply-for-a-juggling-licence")
+    expect(page.find("h1")).to have_text pages[0].question_text
+    expect_page_to_have_no_axe_errors(page)
+
+    fill_in pages[0].question_text, with: Faker::Lorem.sentence
```
Comment (aliuk2012) [inline spec/features/email_confirmation_spec.rb:49]: Minor comment, is it work maybe testing that this value is played back on the CYA page (kind of tests that at least the data is being stored and played back to the user.

**Review by aliuk2012 — APPROVED** [review]
Only a minor suggestion but apart from that it looks good

File: spec/features/email_confirmation_spec.rb (hunk 3 above)
Comment (DavidBiddle) [reply spec/features/email_confirmation_spec.rb]: Thanks @aliuk2012 , have added a check for that and the question text

**Review by aliuk2012 — APPROVED** [review]

**Pull request merged**